ibtk/src/solvers/impls/PETScLevelSolver.cpp

```diff
@@ -307,6 +314,19 @@ void PETScLevelSolver::deallocateSolverState()
     return;
 } // deallocateSolverState
 
+void PETScLevelSolver::addLinearOperator(Mat& op, MatStructure nonzero_pattern)
+{
+#if !defined(NDEBUG)
+	TBOX_ASSERT(!d_is_initialized);
+	TBOX_ASSERT(op);
+#endif
+
+	d_petsc_extern_mat = op;
+	d_extern_mat_nz_pattern = nonzero_pattern;
+
+	return;
+}// addLinearOperator
+
 /////////////////////////////// PROTECTED ////////////////////////////////////
 
 void PETScLevelSolver::init(Pointer<Database> input_db, const std::string& default_options_prefix)
```

Comment:
> On May 2, 2015, at 1:45 PM, Amneet Bhalla <notifications@github.com> wrote:
> 
> In ibtk/src/solvers/impls/PETScLevelSolver.cpp <https://github.com/IBAMR/IBAMR/pull/6#discussion_r29549190>:
> 
> > @@ -307,6 +314,19 @@ void PETScLevelSolver::deallocateSolverState()
> >      return;
> >  } // deallocateSolverState
> >  
> > +void PETScLevelSolver::addLinearOperator(Mat& op, MatStructure nonzero_pattern)
> > +{
> > +#if !defined(NDEBUG)
> > +	TBOX_ASSERT(!d_is_initialized);
> > +	TBOX_ASSERT(op);
> > +#endif
> > +
> > +	d_petsc_extern_mat = op;
> > +	d_extern_mat_nz_pattern = nonzero_pattern;
> > +
> > +	return;
> > +}// addLinearOperator
> > +
> >  /////////////////////////////// PROTECTED ////////////////////////////////////
> >  
> >  void PETScLevelSolver::init(Pointer<Database> input_db, const std::string& default_options_prefix)
> There is no pc_type option here and neither is one used in PETScLevelSolver::initializeSolverState. That should be changed?
> 
If you want to.

Also, please remove addLinearOperator() from the public interface (and change the name to setLinearOperator()).